app/scripts/controllers/visualizerModal.js

```diff
@@ -1,29 +1,37 @@
 'use strict';
 
 angular.module('openhimConsoleApp')
-  .controller('VisualizerModalCtrl', function ($http, $scope, $modalInstance, $timeout, Api, Notify, Alerting, visualizers) {
-
-
-    /***************************************************/
-    /**         Initial page load functions           **/
-    /***************************************************/
+  .controller('VisualizerModalCtrl', function ($http, $scope, $modalInstance, $timeout, Api, Notify, Alerting, visualizers, visualizer, duplicate) {
 
     $scope.ngError = {};
     $scope.validationRequiredMsg = 'This field is required.';
 
     // get/set the users scope whether new or update
-    $scope.update = false;
-
-    // create visualizer settings properties
-    $scope.visualizer = {};
-
-    // load default visualizer config for new user
-    $http.get('config/visualizer.json').success(function( visualizerConfig ) {
-      angular.extend( $scope.visualizer, angular.copy( visualizerConfig ) );
-    });
+    if (visualizer) {
+      $scope.update = true;
+      // make a copy of the object so that the original doesn't get changed until we save
+      $scope.visualizer = JSON.parse(angular.toJson(visualizer));
```

Comment:
Future version of angular added a `toJSON()` method to resource object but the version we are using doesn't have this yet.